system/mp/mpcomm.cpp

```diff
@@ -2178,194 +2265,272 @@ void CMPConnectThread::startPort(unsigned short port)
     if (!listen)
         return;
     running = true;
+    if (acceptThreadPoolSize)
+    {
+        class CFactory : public CInterfaceOf<IThreadFactory>
+        {
+            CMPConnectThread &owner;
+        public:
+            CFactory(CMPConnectThread &_owner) : owner(_owner)
+            {
+            }
+        // IThreadFactory
+            IPooledThread *createNew() override
+            {
+                class CMPConnectionThread : public CInterfaceOf<IPooledThread>
+                {
+                    CMPConnectThread &owner;
+                    Owned<ISocket> sock;
+                public:
+                    CMPConnectionThread(CMPConnectThread &_owner) : owner(_owner)
+                    {
+                    }
+                // IPooledThread
+                    virtual void init(void *param) override
+                    {
+                        sock.set((ISocket *)param);
+                    }
+                    virtual void threadmain() override
+                    {
+                        constexpr unsigned timeoutMs = 5000;
+                        Owned<ISocket> handledSock = sock.getClear();
+                        if (owner.handleAcceptedSocket(handledSock.getLink(), timeoutMs, false))
```

Comment:
Why try for 5000 ms before handing off to async thread(s) ?  Why not just always use async thread(s) ?
Maybe almost same thing if we change 5000 ms here to 100 ms or something very short ..

Reply:
It does always use an async thread - this code is in in CMPConnectionThread::threadmain (a thread in the pool).

What it's doing here (within a thread of the pool), is handing this connection off, to outside the thread pool, to a separate 'slow' handler, so pool doesn't clog up.